packages/rainbowkit/src/components/ConnectButton/ConnectButton.tsx

```diff
@@ -43,8 +43,13 @@ export function ConnectButton({
 
   const i18n = useContext(I18nContext);
 
+  const shouldShowBalance =
+    normalizeResponsiveValue(showBalance)[
+      isMobile() ? 'smallScreen' : 'largeScreen'
+    ];
+
   return (
-    <ConnectButtonRenderer>
+    <ConnectButtonRenderer showBalance={shouldShowBalance}>
```

Comment:
I added an extra prop to the `ConnectButtonRendered` component called `showBalance`. We can pass either `true` or `false` and it will make sure to fetch the balance only if it's `true`.

Reply:
I think we'll want to avoid this addition since it exposes a new prop to the `ConnectButton.Custom` API

Reply:
Not sure the best way to handle this. I wonder if we should create something similar to `ShowRecentTransactionsContext` that's exposed in `RainbowKitProvider` to requested limited RPC calls. That seems like the root cause of the issue; some dApps don't want to pay for excessive RPC usage.